src/withdrawal_limit/tests/withdrawal_limit_test.cairo

```diff
@@ -0,0 +1,222 @@
+    let withdrawal_limit_mock_class_hash = snf::declare("withdrawal_limit_mock").unwrap();
+    let (withdrawal_limit_mock_address, _) = withdrawal_limit_mock_class_hash
+        .deploy(@array![])
+        .unwrap();
+    let withdrawal_limit_mock = IWithdrawalLimitDispatcher {
+        contract_address: withdrawal_limit_mock_address
+    };
+
+    let mut spy = snf::spy_events();
+    (withdrawal_limit_mock, spy)
+}
+
+
+pub fn deploy_erc20(name: ByteArray, symbol: ByteArray) -> ContractAddress {
+    let erc20_class_hash = snf::declare("ERC20").unwrap();
+    let mut constructor_args = ArrayTrait::new();
+    name.serialize(ref constructor_args);
+    symbol.serialize(ref constructor_args);
+    let fixed_supply: u256 = 1000000000;
+    fixed_supply.serialize(ref constructor_args);
+    OWNER().serialize(ref constructor_args);
+
+    let (usdc, _) = erc20_class_hash.deploy(@constructor_args).unwrap();
+
+    let usdc_token = IERC20Dispatcher { contract_address: usdc };
+
+    // Transfering usdc to test address for testing
+    snf::start_cheat_caller_address(usdc, OWNER());
+    usdc_token.transfer(snf::test_address(), 1000_000_000);
+    snf::stop_cheat_caller_address(usdc);
+
+    return usdc;
+}
+
+#[test]
+fn get_remaining_withdrawal_quota_ok() {
+    let (withdrawal_limit, _) = deploy_withdrawal_limit();
+
+    let usdc_address = deploy_erc20("USDC", "USDC");
+    let usdc = IERC20Dispatcher { contract_address: usdc_address };
+    // Mocking deposits with the contract
+    usdc.transfer(withdrawal_limit.contract_address, 1000_000);
+
+    let withdrawal_limit_mock = IMockWithdrawalLimitDispatcher {
+        contract_address: withdrawal_limit.contract_address
+    };
+
+    // Should return 0 when withdrawal limit not applied
+    assert(
+        withdrawal_limit.get_remaining_withdrawal_quota(usdc_address) == BoundedInt::max(),
+        'Quota should be 0'
+    );
+    withdrawal_limit_mock.change_withdrawal_limit_token(usdc_address, true);
+
+    // Should return the default 5% of the balance of contract when limit applied
+    assert(
+        withdrawal_limit.get_remaining_withdrawal_quota(usdc_address) == 5000_0,
+        'Quota should not be 0'
+    );
+}
+
+#[test]
+fn consume_withdrawal_quota_ok() {
+    let (withdrawal_limit, mut spy) = deploy_withdrawal_limit();
+
+    let usdc_address = deploy_erc20("USDC", "USDC");
+    let usdc = IERC20Dispatcher { contract_address: usdc_address };
+    // Mocking deposits with the contract
+    usdc.transfer(withdrawal_limit.contract_address, 1000_000);
+
+    let withdrawal_limit_mock = IMockWithdrawalLimitDispatcher {
+        contract_address: withdrawal_limit.contract_address
+    };
+
+    withdrawal_limit_mock.change_withdrawal_limit_token(usdc_address, true);
+
+    assert(
+        withdrawal_limit.get_remaining_withdrawal_quota(usdc_address) == 5000_0,
+        'Quota should not be 0'
+    );
+    withdrawal_limit_mock.consume_quota(usdc_address, 10000);
```

Comment:
```suggestion
    withdrawal_limit_mock.consume_quota(usdc_address, 10_000);
```
I believe the normal `_` practice is every 3 digits from the right. or, we can skip `_` and use smaller numbers like 100 as mentioned above